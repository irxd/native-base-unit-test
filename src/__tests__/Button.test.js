import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import {
  NativeBaseProvider, Button
} from 'native-base';

describe('<Button />', () => {
  it('should render button component properly', () => {
    const { queryByTestId } = render(
      <NativeBaseProvider
        initialWindowMetrics={{
          frame: { x: 0, y: 0, width: 0, height: 0 },
          insets: { top: 0, left: 0, right: 0, bottom: 0 },
        }}
      >
        <Button testID="my-button">Test</Button>
      </NativeBaseProvider>
    );

    expect(queryByTestId('my-button')).not.toBeNull()
  });

  it('should render button props properly', () => {
    const { queryByTestId } = render(
      <NativeBaseProvider
        initialWindowMetrics={{
          frame: { x: 0, y: 0, width: 0, height: 0 },
          insets: { top: 0, left: 0, right: 0, bottom: 0 },
        }}
      >
        <Button testID="my-button" backgroundColor="white" borderRadius={8}>Test</Button>
      </NativeBaseProvider>
    );

    expect(queryByTestId('my-button').props.style.backgroundColor).toMatch('#FFFFFF');
    expect(queryByTestId('my-button').props.style.borderRadius).toEqual(8);
  });

  it('button event should working properly', () => {
    const onPressEvent = jest.fn();

    const { queryByTestId } = render(
      <NativeBaseProvider
        initialWindowMetrics={{
          frame: { x: 0, y: 0, width: 0, height: 0 },
          insets: { top: 0, left: 0, right: 0, bottom: 0 },
        }}
      >
        <Button testID="my-button" onPress={onPressEvent}>Test</Button>
      </NativeBaseProvider>
    );

    fireEvent.press(queryByTestId('my-button'));
    expect(onPressEvent).toHaveBeenCalled();
  });
});
