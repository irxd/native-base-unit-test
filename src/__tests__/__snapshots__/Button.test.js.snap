// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> button event should working properly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    _text={
      Object {
        "color": "lightText",
      }
    }
    accessibilityRole="button"
    accessible={true}
    collapsable={false}
    colorScheme="primary"
    dataSet={Object {}}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onHoverIn={[Function]}
    onHoverOut={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#06b6d4",
        "borderRadius": 4,
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingBottom": 8,
        "paddingLeft": 12,
        "paddingRight": 12,
        "paddingTop": 8,
      }
    }
    testID="my-button"
  >
    <View
      dataSet={Object {}}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <View
        dataSet={Object {}}
        style={Object {}}
      >
        <Text
          dataSet={Object {}}
          style={
            Object {
              "backgroundColor": undefined,
              "color": "#FFFFFF",
              "fontFamily": undefined,
              "fontSize": 14,
              "fontStyle": "normal",
              "fontWeight": "500",
              "letterSpacing": 0,
              "lineHeight": 21,
              "textDecorationLine": undefined,
            }
          }
        >
          Test
        </Text>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
